ikarus/finiteelements/fehelper.hh

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+#include <dune/localfefunctions/manifolds/realTuple.hh>
+
+#include <ikarus/finiteelements/ferequirements.hh>
+#include <ikarus/finiteelements/physicshelper.hh>
+
+namespace Ikarus::FEHelper {
+  /**
+   * @brief Gets the local displacement Dune block vector
+   *
+   * @tparam LocalView Type of the local view.
+   * @tparam FERequirementType Type of the FE requirements.
+   * @tparam ScalarType Scalar type for the local displacement vector.
+   *
+   * @param localView Local view of the element.
+   * @param par The FERequirementType object.
+   * @param dx Optional displacement vector.
+   *
+   * @return A Dune block vector representing the displacement quantities at each node.
+   * */
+  template <typename LocalView, typename FERequirementType, typename ScalarType>
+  auto localDisplacementBlockVector(const LocalView& localView, const FERequirementType& par,
+                                    const std::optional<const Eigen::VectorX<ScalarType>>& dx = std::nullopt) {
+    using Traits           = TraitsFromLocalView<LocalView>;
+    constexpr int worldDim = Traits::worlddim;
+    const auto& d          = par.getGlobalSolution(Ikarus::FESolutions::displacement);
```

Comment:
I would give here the vector d as argument and the localView. Then it is not displacement specific. You can the us a more generic name.